cypress/e2e/webTables.cy.js

```diff
@@ -1,5 +1,126 @@
+/// <reference types="cypress" />
+
 describe('Web Tables page', () => {
-  it('', () => {
+  function deleteAllFoundWorkers(rawCount) {
+    if (rawCount < 1) return; // if no rows remain, terminate recursion
+    cy.get('[id^="delete-record"]', { timeout: 10000 })
+      .first()
+      .click()
+      .then(() => {
+        deleteAllFoundWorkers(--rawCount); // try again
+      });
+  }
+  
+  const searchData = {
+    firstName: 'Cierra',
+    lastName: 'Vega',
+    age: '39',
+    email: 'cierra@example.com',
+    salary: '10000',
+    department: 'Insurance',
+  };
+
+  const testData = {
+    newWorker: {
+      firstName: 'Anton',
+      lastName: 'Kalnohuz',
+      email: 'akalnohuz@example.com',
+      age: '25',
+      salary: '50000',
+      department: 'IT'
+    }
+  };
+
+  beforeEach(() => {
+    cy.visit('https://demoqa.com/webtables');
+  });
+
+  it('pagination should be present', () => {
+    cy.get('.-pagination').should('exist');
+  });
+
+  it('rows count selection should be present and working', () => {
+    cy.get('.select-wrap.-pageSizeOptions').should('exist');
+    cy.get('.select-wrap.-pageSizeOptions').click();
+    cy.get('[aria-label="rows per page"]').select('5 rows');
+    cy.contains('[aria-label="rows per page"]', '5 rows');
+  });
+
+  it('should add a new worker', () => {
+    cy.get('#addNewRecordButton').click();
+
+    cy.get('#firstName').type(testData.newWorker.firstName);
+    cy.get('#lastName').type(testData.newWorker.lastName);
+    cy.get('#userEmail').type(testData.newWorker.email);
+    cy.get('#age').type(testData.newWorker.age);
+    cy.get('#salary').type(testData.newWorker.salary);
+    cy.get('#department').type(testData.newWorker.department);
+
+    cy.contains('button', 'Submit').click();
+  });
+
+  it("Should be possible to delete all workers", () => {
+    cy.visit("https://demoqa.com/webtables");
+    cy.get('[id ^= "delete-record"]').then((rows) =>
+      deleteAllFoundWorkers(rows.length)
+    );
+    // Assert that table is empty
+    cy.contains("No rows found").should("be.visible");
+  });
+
+
+
+  it('should find and edit a worker', () => {
+    cy.get('#searchBox').type('Cierra');
```

Comment:
make sure that Cierra was found

Reply:
it seems to me that cierra was found. https://imgur.com/a/aqv8q1p

Reply:
I mean you should add an assertion before editing

Reply:
OK, i get it. )